docs/userguide.md

```diff
@@ -0,0 +1,325 @@
+
+### Deploy the Cluster Registry
+
+Make sure the storage provisioner is ready before deploying the Cluster
+Registry.
+
+```bash
+kubectl -n kube-system get pod storage-provisioner
+```
+
+Get version 0.0.4 of the [Cluster
+Registry](https://github.com/kubernetes/cluster-registry/releases/tag/v0.0.4) and run the
+following:
+
+```bash
+crinit aggregated init mycr --host-cluster-context=cluster1
+```
+
+You can also specify your own cluster registry image using the `--image` flag.
+
+### Deploy Federation
+
+First you'll need to create the namespace to deploy the federation into:
+
+```bash
+kubectl create ns federation
+```
+
+Then run `apiserver-boot` to deploy. This will build the federated `apiserver`
+and `controller`, build an image containing them, push the image to the
+requested registry, generate a k8s YAML config `config/apiserver.yaml`
+with the name and namespace requested, and deploy the config to the
+cluster specified in your `kubeconfig config current-context`.
+
+If intending to use the docker hub as the container registry to push
+the federation image to, make sure to login to the local docker daemon
+to ensure credentials are available for push:
+
+```bash
+docker login --username <username>
+```
+
+Once you're ready, run the following command. An example of the `--image`
+argument is `docker.io/<username>/federation-v2:test`.
+
+```bash
+apiserver-boot run in-cluster --name federation --namespace federation \
+    --image <containerregistry>/<username>/<imagename>:<tagname> \
+    --controller-args="-logtostderr,-v=4"
+```
+
+You will most likely need to increase the limit on the amount of memory
+required by the `apiserver` as there isn't a way to request it via
+`apiserver-boot` at the moment. Otherwise the federation pod may be terminated
+with `OOMKilled`. In order to do that you will need to patch the deployment:
+
+```bash
+kubectl -n federation patch deploy federation -p \
+    '{"spec":{"template":{"spec":{"containers":[{"name":"apiserver","resources":{"limits":{"memory":"128Mi"},"requests":{"memory":"64Mi"}}}]}}}}'
+```
+
+If you also run into `OOMKilled` on either the `etcd` or the `controller`, just
+run the same command but replace `apiserver` with `etcd` or `controller`.
+
+Verify that the deployment succeeded and is available to serve its API by
+seeing if we can retrieve one of its API resources:
+
+```bash
+kubectl get federatedcluster
+```
+
+It should successfully report that no resources are found.
+
+### Enabling Push Propagation
+
+Configuration of push propagation is via the creation of
+FederatedTypeConfig resources. To enable propagation for default
+supported types, run the following command:
+
+```bash
+for tc in ./config/federatedtypes/*.yaml; do kubectl create -f "${tc}"; done
+```
+
+Once this is complete, you now have a working federation-v2 control plane and
+can proceed to join clusters.
+
+### Join Clusters
+
+Next, you'll want to use the `kubefnord` tool to join all your
```

Comment:
Lets give this tool a better name.  
I've opened https://github.com/kubernetes-sigs/federation-v2/issues/103 to track that.
We can merge this doc as is, and update it once the above is done.